src/nomad_simulations/schema_packages/basis_set.py

```diff
@@ -190,25 +190,170 @@ class AtomCenteredFunction(ArchiveSection):
+                )
+
+        # Resolve combined types
+        if self.function_type and len(self.function_type) > 1:
+            num_types = len(self.function_type)  # For SP: 2, SPD: 3, etc.
+            if self.contraction_coefficients is not None:
+                expected_coeffs = num_types * self.n_primitive
+                if len(self.contraction_coefficients) != expected_coeffs:
+                    raise ValueError(
+                        f'Mismatch in contraction coefficients for {self.function_type} type: '
+                        f'expected {expected_coeffs}, found {len(self.contraction_coefficients)}.'
+                    )
+
+                # Split coefficients into separate lists for each type
+                self.coefficient_sets = {
+                    t: self.contraction_coefficients[i::num_types]
+                    for i, t in enumerate(self.function_type)
+                }
+
+                # Debug: Log split coefficients
+                for t, coeffs in self.coefficient_sets.items():
+                    logger.info(f'{t}-type coefficients: {coeffs}')
+            else:
+                logger.warning(
+                    f'No contraction coefficients provided for {self.function_type} type.'
+                )
+
+        # For single types, ensure coefficients match primitives
+        elif self.contraction_coefficients is not None:
+            if len(self.contraction_coefficients) != self.n_primitive:
+                raise ValueError(
+                    f'Mismatch in contraction coefficients: expected {self.n_primitive}, '
+                    f'found {len(self.contraction_coefficients)}.'
+                )
 
 
 class AtomCenteredBasisSet(BasisSetComponent):
     """
     Defines an atom-centered basis set.
     """
 
+    basis_set = Quantity(
+        type=str,
+        description="""
+        name of the basis set.
+        """,
+    )
+
+    type = Quantity(
+        type=MEnum(
+            'STO',  # Slater-type orbitals
+            'GTO',  # Gaussian-type orbitals
+            'NAO',  # Numerical atomic orbitals
+            'cECP',  # Capped effective core potentials
+            'PC',  # Point charges
+        ),
+        description="""
+        Type of the basis set, e.g. STO or GTO.
+        """,
+    )
+
+    role = Quantity(
+        type=MEnum(
+            'orbital',
+            'auxiliary_scf',
+            'auxiliary_post_hf',
+            'cabs',  # complementary auxiliary basis set
+        ),
+        description="""
+        The role of the basis set.
+        """,
+    )
+
+    total_number_of_basis_functions = Quantity(
+        type=np.int32,
+        description='',
+    )
+
     functional_composition = SubSection(
         sub_section=AtomCenteredFunction.m_def, repeats=True
-    )  # TODO change name
+    )
 
     def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
         super().normalize(archive, logger)
         # self.name = self.m_def.name
```

Comment:
Don't forget to do something with this.

Do we already have some sort of convention the names? It seems like most or at least a lot of classes have the line you are commenting out here, but I have also seen `self.name = self.m_def.name if self.name is None else self.name`  which allows the parser to overwrite this name.